kong/plugins/jwt/handler.lua

```diff
@@ -56,13 +56,20 @@ function JwtHandler:access(conf)
     return responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
   end
 
-  if not token then
-    return responses.send_HTTP_UNAUTHORIZED()
+  if type(token) ~= "string" then
+    if type(token) == "nil" then
+      return responses.send_HTTP_UNAUTHORIZED()
+    elseif type(token) == "table" then
+      return responses.send_HTTP_UNAUTHORIZED("Multiple tokens provided")
+    else
+      return responses.send_HTTP_UNAUTHORIZED("Unrecognizable token")
+    end
   end
-
+  
   -- Decode token to find out who the consumer is
   local jwt, err = jwt_decoder:new(token)
   if err then
+    -- TODO: is this an internal server error? the token is invalid/cannot be parsed so isn't this a user error to be reported?
```

Comment:
@thibaultCha ^^

The possible errors [originate here](https://github.com/Mashape/kong/blob/74940cfbc43bd5eae25bdd86cbde664ae6294adb/kong/plugins/jwt/jwt_parser.lua#L99-L117)

They seem user errors to me, any specific reason they are set to fail with a 500?